e2e/jenkins/crw-ci/k8s/pr-check/Jenkinsfile

```diff
@@ -0,0 +1,229 @@
+  /usr/local/bin/chectl server:start \\
+   --k8spodreadytimeout=180000 \\
+   -t=${WORKSPACE}/deploy/ \\
+   --listr-renderer=verbose \\
+   --cheimage=${CHE_IMAGE_REPO}:${CHE_IMAGE_TAG}
+"""
+
+                    // wait che-server to be available
+                    sh """
+     CHE_URL=\$(kubectl get ingress che-ingress -n=che -o=jsonpath={'.spec.rules[0].host'})
+     
+     COUNTER=0;
+     SUCCESS_RATE_COUNTER=0;
+     while true; do
+      if [ \$COUNTER -gt 180 ]; then
+      echo "Unable to get stable route. Exiting"
+        exit 1
+      fi
+      
+      ((COUNTER+=1))
+      
+      
+      STATUS_CODE=\$(curl -sL -w "%{http_code}" -I \${CHE_URL} -o /dev/null; true) || true # added true to not fail the script completely.
+      
+      echo "Try \${COUNTER}. Status code: \${STATUS_CODE}"
+      if [ "\$STATUS_CODE" == "200" ]; then 
+        ((SUCCESS_RATE_COUNTER+=1))
+      fi
+      sleep 1;
+    
+      if [ \$SUCCESS_RATE_COUNTER == \$SUCCESS_THRESHOLD ]; then 
+        echo "Route is stable enough. Continuing running tests"
+        break
+      fi
+     done
+"""
+                }
+            }
+        }
+
+        stage("Create ws from devfile") {
+            steps {
+                script {
+                    sh """
+  /usr/local/bin/chectl workspace:start --devfile=$DEVFILE_URL
+  kubectl get configmaps --namespace=che che -o yaml
+"""
+                }
+            }
+        }
+
+        stage("Run Happy path tests against Che on Kubernetes") {
+            steps {
+                script {
+                    // TODO switch to eclipse/che-e2e image
+//                    sh """
+// CHE_HOST=\$(kubectl get ingress che-ingress -n=che -o=jsonpath={'.spec.rules[0].host'})
+// CHE_URL=http://\${CHE_HOST}
+// docker run --shm-size=256m --net=host --ipc=host \\
+//   -e TS_SELENIUM_HEADLESS='true' \\
+//   -e TS_SELENIUM_DEFAULT_TIMEOUT=300000 \\
+//   -e TS_SELENIUM_LOAD_PAGE_TIMEOUT=240000 \\
+//   -e TS_SELENIUM_WORKSPACE_STATUS_POLLING=20000 \\
+//   -e TS_SELENIUM_BASE_URL=\${CHE_URL} \\
+//   -v ${WORKSPACE}/che/e2e:/root/local_tests:Z \\
+//   eclipse/che-e2e:nightly
+//"""
+
+                      sh """
+ CHE_HOST=\$(kubectl get ingress che-ingress -n=che -o=jsonpath={'.spec.rules[0].host'})
+ CHE_URL=http://\${CHE_HOST}
+ docker run --net=host --ipc=host \\
+     -e TS_SELENIUM_HEADLESS='true' \\
+     -e TS_SELENIUM_DEFAULT_TIMEOUT=300000 \\
+     -e TS_SELENIUM_LOAD_PAGE_TIMEOUT=240000 \\
+     -e TS_SELENIUM_BASE_URL=\${CHE_URL} \\
+     -e TS_SELENIUM_WORKSPACE_STATUS_POLLING=20000 \\
+     -w /home/che/e2e \\
+     -v $WORKSPACE:/home:Z \\
+     cypress/browsers:node8.9.3-chrome73 bash -c "npm install &&  npm run test-happy-path"
+"""
+                }
+            }
+        }
+    }
+
+    post {
+        always {
+            script {
+                // it's impossible to use 'when' in the global post block
+                if (currentBuild.currentResult != 'ABORTED') {
```

Comment:
Why is this not executing on aborting the job?

Reply:
Agree. Let's execute it anyway. Thanks for advice!